ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Wiki/WikiPagesSample.cs

```diff
@@ -0,0 +1,285 @@
+
+            List<WikiV2> wikis = wikiClient.GetAllWikisAsync(projectId).SyncResult();
+
+            if (wikis == null || wikis.Count == 0)
+            {
+                Console.WriteLine("No wikis exist to get wiki page");
+
+                return null;
+            }
+
+            WikiV2 wiki = wikis[0];
+            WikiPageResponse rootPageResponse = wikiClient.GetPageAsync(
+                project: wiki.ProjectId,
+                wikiIdentifier: wiki.Name,
+                path: "/",
+                recursionLevel: VersionControlRecursionType.OneLevel).SyncResult();
+
+            Console.WriteLine("Retrieved the following subpages for the root page:");
+            foreach (WikiPage subPage in rootPageResponse.Page.SubPages)
+            {
+                Console.WriteLine("Sub-page : '{0}'", subPage.Path);
+            }
+
+            return rootPageResponse;
+        }
+
+        [ClientSampleMethod]
+        public string GetWikiPageText()
+        {
+            VssConnection connection = this.Context.Connection;
+            WikiHttpClient wikiClient = connection.GetClient<WikiHttpClient>();
+
+            Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+
+            List<WikiV2> wikis = wikiClient.GetAllWikisAsync(projectId).SyncResult();
+
+            if (wikis == null || wikis.Count == 0)
+            {
+                Console.WriteLine("No wikis exist to get wiki page");
+
+                return null;
+            }
+
+            WikiV2 wiki = wikis[0];
+            WikiPage rootPage = wikiClient.GetPageAsync(
+                project: wiki.ProjectId,
+                wikiIdentifier: wiki.Id,
+                path: "/",
+                recursionLevel: VersionControlRecursionType.OneLevel).SyncResult().Page;
+
+            string somePagePath = rootPage.SubPages[0].Path;
+            
+            using (var reader = new StreamReader(wikiClient.GetPageTextAsync(
+                project: wiki.ProjectId,
+                wikiIdentifier: wiki.Id,
+                path: somePagePath).SyncResult()))
+            {
+                string pageContent = reader.ReadToEnd();
+                Console.WriteLine("Retrieved page '{0}' in wiki '{1}' with content '{2}'", somePagePath, wiki.Name, pageContent);
+
+                return pageContent;
+            }
+        }
+
+        [ClientSampleMethod]
+        public WikiPageResponse EditWikiPage()
+        {
+            VssConnection connection = this.Context.Connection;
+            WikiHttpClient wikiClient = connection.GetClient<WikiHttpClient>();
+
+            Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+
+            List<WikiV2> wikis = wikiClient.GetAllWikisAsync(projectId).SyncResult();
+
+            if (wikis == null || wikis.Count == 0)
+            {
+                Console.WriteLine("No wikis exist to get wiki page");
+
+                return null;
+            }
+
+            WikiV2 wiki = wikis[0];
```

Comment:
The logic to get a random page to edit is somewhat distracting and makes it harder to discover the important part of this sample (the call to get a page with its content and update it). You might consider a util method to find a random page (e.g. `WikiSampleHelpers.FindAnyWikiPage`) OR just have this sample update the root page.